pennylane/operation.py

```diff
@@ -891,6 +891,18 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
                 f"{len(self._wires)} wires given, {self.num_wires} expected."
             )
 
+        # Check that the wires is not an empty list for operators that have
+        # `num_wires` as `AnyWires` or `AllWires`.
+        if self.num_wires in {AllWires, AnyWires}:
+            if not isinstance(self, (qml.Barrier, qml.Snapshot, qml.Hamiltonian)):
```

Comment:
Is there are test cases already for each of these operations testing that no error is raised when passing e.g., an empty list as wires?

Reply:
That's a good point! I have now added test cases for `qml.Barrier`, `qml.Snapshot`, `qml.Hamiltonian` testing that no 'empty wire list' error is raised for these cases.